packages/amplify-graphql-transformer-core/src/transformer-context/index.ts

```diff
@@ -98,4 +101,12 @@ export class TransformerContext implements TransformerContextProvider {
   public isProjectUsingDataStore(): boolean {
     return !!this.resolverConfig?.project || !!this.resolverConfig?.models;
   }
+
+  public isProjectUsingCPK = (): boolean => {
+    const obj = this.inputDocument.definitions.find(def => def.kind === 'ObjectTypeDefinition') as ObjectTypeDefinitionNode;
+    const primaryKeyField = obj.fields?.find(field => field.directives?.find(directive => directive.name.value === 'primaryKey'));
+    if (!primaryKeyField || primaryKeyField.name.value === DEFAULT_ID_FIELD_NAME) {
```

Comment:
I think the following schema should also be considered as custom primary key (id as primary key and with sort key fields defined)
```GraphQL
type Todo @model {
  id: ID! @primaryKey(sortKeyFields:["name"])
  name: String!
}
```

Reply:
+1